packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/dto/ConnectorTool.java

```diff
@@ -0,0 +1,27 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.collaborative.diagrams.dto;
+
+import org.eclipse.sirius.components.representations.IStatus;
+import org.eclipse.sirius.components.representations.VariableManager;
+
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * DTO for the connector tools.
+ *
+ * @author mcharfadi
+ */
+public record ConnectorTool(String id, String label, List<String> iconURL, Function<VariableManager, IStatus> handler, List<String> candidatesDescriptionId) {
```

Comment:
This object is a DTO so it's supposed to be sent over the network. Contrary to the previous tool DTOs which were mixing responsibilities. This one should never in any way include something like this handler. This is what we are trying to remove from the previous tools. On top of that, you don't need it.